File: app/src/main/java/org/lineageos/updater/UpdatesListActivity.java
/*
 * SPDX-FileCopyrightText: The LineageOS Project
 * SPDX-License-Identifier: Apache-2.0
 */
package org.lineageos.updater;

import androidx.appcompat.app.AppCompatActivity;

import org.lineageos.updater.data.Update;

public abstract class UpdatesListActivity extends AppCompatActivity {
    public abstract void exportUpdate(Update update);
    public abstract void showSnackbar(int stringId, int duration);
}

File: app/src/main/java/org/lineageos/updater/UpdatesListActivity.kt
/*
 * SPDX-FileCopyrightText: 2017-2023 The LineageOS Project
 * SPDX-License-Identifier: Apache-2.0
 */

package org.lineageos.updater

import androidx.appcompat.app.AppCompatActivity
import org.lineageos.updater.model.UpdateInfo

abstract class UpdatesListActivity : AppCompatActivity() {
    abstract fun exportUpdate(update: UpdateInfo)
    abstract fun showSnackbar(stringId: Int, duration: Int)
}
